### PR TITLE
Mac-friendly KBD 75% layout with Colemak and Dvorak layers

### PR DESCRIPTION
**Pull request by aaronireland**
## Description

This layout is a KBD 75% 1u ANSI layout which includes a few macros/keymaps for the Mabook hotbar as well as Colemak and Dvorak layers. I noticed that there isn't currently a KBD 75 keymap in the main branch that has layers for both so I thought I'd submit mine. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

*  N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    _______,          _______, _______, COLEMAK, BL_DEC , BL_TOGG, BL_INC , BL_STEP, _______, _______, _______, _______,          KC_PAUS, KC_MUTE,
+    KC_LGUI, KC_LALT, KC_LCTL,                   KC_SPC ,                                     KC_MNXT, KC_MPLY, _______, KC_VOLD, KC_SLCK, KC_VOLU
+  ),
+};
+
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:175]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:181]:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_COLEMAK);
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:187]:
```suggestion
        set_single_persistent_default_layer(_COLEMAK);
```

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_DVORAK);
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:193]:
```suggestion
        set_single_persistent_default_layer(_DVORAK);
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c (hunk 1 above)
Comment (aaronireland) [reply keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c]: Done. Thanks for the review!

File: .gitignore [hunk 5]
```diff
@@ -71,6 +71,7 @@ id_rsa_*
 
 # python things
 __pycache__
+.venv
```
Comment (drashna) [inline .gitignore:74]: Could this be reverted? 

If it needs to be added, it should be done as a separate PR.

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 6]
```diff
@@ -0,0 +1,194 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+// KBD75 layer indices
+#define _QWERTY 0
+#define _COLEMAK 1
+#define _DVORAK 2
+#define _FL 3
```
Comment (drashna) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/keymap.c:7]:
```suggestion
enum custom_layers {
    _QWERTY,
    _COLEMAK,
    _DVORAK,
    _FL,
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/kbd75/keymaps/aaronireland/README.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# Aaron's KBD75v2 Mac-friendly ANSI 1u Layout
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd75/keymaps/aaronireland/README.md:1]: Please rename this file to lowercase.

Comment (aaronireland) [reply keyboards/kbdfans/kbd75/keymaps/aaronireland/README.md]: ok, done. Thanks for the review!

**Review by fauxpark — APPROVED** [review]
